### ISSUE TITLE
Request the ability to embed <image> tags in the card text field (inside cards.xml)

### ISSUE THREAD

**Issue by Qriist** [Cockatrice/Cockatrice#3040]
Long and short, I am looking for the ability to tag a picture inside the <text> field in the cards.xml (and the TK.xml, and custom sets).

The implementation would require a certain amount of extra parsing and requests for images, of course, but should be fairly minimal overall, especially if implemented with a "save local copy" aspect like the card pictures have. Here is an example, using the card Shattergang Brothers.

Original text, as seen now:
`{2}{B}, Sacrifice a creature: Each other player sacrifices a creature.`
`{2}{R}, Sacrifice an artifact: Each other player sacrifices an artifact.`
`{2}{G}, Sacrifice an enchantment: Each other player sacrifices an enchantment.`

Modified text with new parser:
`<image Alt="{2}" src="http://some.server/Mana_2.png"><image Alt="{B}"  src="http://some.server/Mana_B.png">, Sacrifice a creature: Each other player sacrifices a creature.`
`<image Alt="{2}" src="http://some.server/Mana_2.png"><image Alt="{R}"  src="http://some.server/Mana_R.png">, Sacrifice an artifact: Each other player sacrifices an artifact.`
`<image Alt="{2}" src="http://some.server/Mana_2.png"><image Alt="{G}"  src="http://some.server/Mana_G.png">, Sacrifice an enchantment: Each other player sacrifices an enchantment.`

Note the Alt attribute, this serves two functions. In the card text box on the right-hand side, you can mouseover the individual picture and get the text representation (like a normal hover text.) Also, though, cockatrice could display the Alt text instead of the picture if both the local and remote pictures cannot be found.

Finally, if this is implemented, consider also including a decoder for base64 strings for the src url. That would look like `src="data:img/png;base64,{BASE64 STRING HERE}"` and would allow for directly embedding pictures into the xml. This portion could perhaps be declared in the xml so it only needs to be put into the xml once. Perhaps an over all <Images> section, to be referenced like: `src="$Mana_2`

In summary:
-Add ability to render pictures inline with the text
-Possibly add base64 support to enable transmitting of pictures directly in the xml (benefits custom xmls, primarily)

**Comment by ctrlaltca** [comment]
The idea is nice but the proposed implementation doesn't sound really good to me, since it will add a lot of duplicated text inside the cards.xml; eg. if every `{T}` gets replaced with a `<image Alt="{T}" src="data:img/png;base64,<some kb of data here>">`.

What are possible alternatives? Maybe we could add a `symbols` section where special combinations of characters are associated to an image (think of emoticons), eg:
```
<symbols>
  <symbol picURL="http://...">{T}</symbol>
// or even
  <symbol>
    <text>{T}</text>
    <picURL>http://...</picURL>
    <picData>base64data</picData>
  </symbol>
</symbols>
```
And then dynamically replace text with associated images when is shown.

**Comment by Qriist** [comment]
Not a bad idea. Takes the good of mine and removes the need to mass-edit the xml. Cheers!

**Comment by Qriist** [comment]
With the recent new xml engine, how close do you think this is?

**Comment by ctrlaltca** [comment]
It's possible, but still not implemented.

**Comment by ZeldaZach** [comment]
I don't particularly like this idea. Might open up vectors of attack we're not considering.
